[jrchoo] iP

jrchoo opened the pull request:
# Gideon

> "The key is not to prioritise what's on your schedule, but to schedule your priorities." - Stephen Covey

Gideon frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/jrchoo/ip/releases/tag/A-Release).
2. double-click it.
3. add your tasks
4. let it manage your tasks for you 👏 

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jrchoo committed:
initial skeletal version and rename

jrchoo committed:
getting the bot to echo

jrchoo committed:
added ability to store history in a list

jrchoo committed:
added a task class with the ability to mark tasks as done

jrchoo committed:
added 3 classes, todo, deadline and event that extend from task

jrchoo committed:
input and expected output

jrchoo committed:
created custom exception classes but unable to implement yet, cuurently able to handle the basic errors

jrchoo committed:
delete functionality

jrchoo committed:
Add a way to store data via the java File import

jrchoo committed:
Add read and write capabilties to chatbot

jrchoo committed:
Update chatbot so that file is correctly overwritten at the end

jrchoo committed:
Merge branch 'b1'

jrchoo committed:
Add LocalDateTime in order to store deadlines as date and time

jrchoo committed:
Merge branch 'branch-Level-8'

jrchoo committed:
Add Ui, Storage, TaskList and Parser classes

jrchoo committed:
Separate main method logic into the classes

jrchoo committed:
Merge branch 'branch-A-MoreOOP'

jrchoo committed:
Add enums to contain valid commands

jrchoo committed:
Merge branch 'branch-A-MoreOOP'

jrchoo committed:
Create duke package

jrchoo committed:
Merge branch 'branch-A-Packages'

jrchoo committed:
Merge branch 'add-gradle-support'

jrchoo committed:
Update build.gradle to work correctly

jrchoo committed:
Add JUnit tests for TaskList and Todo

jrchoo committed:
Add new constructor for TaskList

jrchoo committed:
Merge branch 'branch-A-JUnit'

DistractedCat commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,21 @@
+public class Event extends Task {
+
+    protected String from;
+    protected String to;
```
Good use of visibility markers.

DistractedCat commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        } else {
+            System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+        return false;
```
might be better to consider using a switch statement.

DistractedCat commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,75 @@
+        return tasks;
+    }
+
+    public void saveTasks(TaskList taskList) {
```
good use of verbs to describe function names.

DistractedCat commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,86 @@
+import java.util.Scanner;
+
+public class Ui {
```
Good abstraction of ui into its separate class.

DistractedCat reviewed (commented):
LGTM!

jrchoo committed:
Add JavaDoc comments to the code

jrchoo committed:
Merge branch 'branch-A-JavaDoc'

cheeggered commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,114 @@
+     * @param command The user's input command.
+     * @return True if the command was successfully parsed and executed, false otherwise.
+     */
+    public boolean parseCommand(String command) {
```
Perhaps you can consider making Command classes and have the boolean isExit as an attribute

cheeggered commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,114 @@
+            }
+        }
+        if (cmd.equals(Commands.bye)) {
+            ui.byeMessage();
```
For OOP, is it the responsibility of the parser class to print these messages?

cheeggered commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,114 @@
+                ui.markedMessage(taskList.getTask(id - 1));
+                return true;
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("To mark a task you need to include the index");
```
Good job dealing with other types of errors!

cheeggered commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,100 @@
+package duke;
+public class Parser {
+
+    private enum Commands {
```
Good attempt at utilising enums!

jrchoo committed:
Remove duplicate classes

jrchoo committed:
Add JavaDoc comments for duke package

jrchoo committed:
Merge branch 'branch-A-JavaDoc'

jrchoo committed:
Add find functionality to the chatbot

jrchoo committed:
Merge branch 'branch-Level-9'

jrchoo committed:
Add GUI to chatbot

jrchoo committed:
Merge branch 'branch-Level-10'

jrchoo committed:
Include assertion for key methods

jrchoo committed:
Add assertions to Storage and TaskList

jrchoo committed:
Refactor parser logic into separate command classes

jrchoo committed:
Add several duke exception classes

jrchoo committed:
Add JavaDoc comments for the newly created classes

jrchoo committed:
Refactor and clean up codebase

Reorganise the Parser class for better readability.
- Main logic of the parseCommand method has been extracted into various
parse methods.
- The parse methods parse the user input into suitable formats which
are then used to create command objects.

Extract the common behavior of Command classes (e.g. the execute method)
into a super class to allow us to use polymorphism when dealing with
the respective commands.

Improve error handling with the use of DukeException and its subclasses.

Additional changes made.
1. Remove unused imports and variables.
2. Update comments for clarity and consistency.

jrchoo committed:
Merge pull request #1 from jrchoo/branch-A-Assertions

Include assertion for key methods

jrchoo committed:
Merge branch 'master' of https://github.com/jrchoo/ip

* 'master' of https://github.com/jrchoo/ip:
  Add assertions to Storage and TaskList
  Include assertion for key methods

jrchoo committed:
Resolve merge conflict in Parser.java

jrchoo committed:
Merge pull request #3 from jrchoo/branch-A-CodeQuality

Improve code quality

jrchoo committed:
Merge branch 'master' of https://github.com/jrchoo/ip

* 'master' of https://github.com/jrchoo/ip:
  Refactor and clean up codebase
  Add JavaDoc comments for the newly created classes
  Add several duke exception classes
  Refactor parser logic into separate command classes

jrchoo committed:
Add a method to edit description of a task

jrchoo committed:
Add a method to parse edit commands from the user

jrchoo committed:
Add a method for the ui to display messages for the edit command

jrchoo committed:
Create a EditCommand class that can be executed

jrchoo committed:
Add task edit feature

Description:
- Implemented a new feature that allows users to edit the description
of a task.
- Added the EditCommand class to handle edit operations.
- Modified the Task class to include a method for editing descriptions.
- Updated the Parser class to recognize and parse the "edit" command.
- Added user feedback messages in the Ui class for successful edits.

Usage:
- Users can now edit the description of an existing task using the
"edit" command.
- Example: `edit 1 /to newDescription` will edit the description
of task 1.

Files Modified:
- Added EditCommand class
- Modified Task class
- Updated Parser class
- Updated Ui class

Notes:
- This feature enhances the usability of the Duke application by
allowing users to make changes to existing tasks.
- Further enhancements can include the ability to edit task dates
or other attributes.

jrchoo committed:
Merge pull request #4 from jrchoo/C-Update

Implement extension: C-Update

jrchoo committed:
Merge branch 'master' of https://github.com/jrchoo/ip

* 'master' of https://github.com/jrchoo/ip:
  Add task edit feature
  Create a EditCommand class that can be executed
  Add a method for the ui to display messages for the edit command
  Add a method to parse edit commands from the user
  Add a method to edit description of a task

jrchoo committed:
Fix GUI to display messages upon entering and exiting

jrchoo committed:
Merge branch 'branch-GUI-tweaks'

jrchoo committed:
Add a user guide

jrchoo committed:
Merge branch 'branch-A-UserGuide'

jrchoo committed:
Add screenshot of Ui

jrchoo committed:
Merge branch 'branch-A-UserGuide'
